.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,404 @@
+
+      - name: Set Tiller URL
+        run: echo "TILLER_URL=https://github.com/rancher/helm/releases/download/${{ env.CATTLE_HELM_VERSION }}/rancher-tiller" >> $GITHUB_ENV
+
+      - name: Download Rancher Helm
+        run: sudo curl -sLf ${{ env.HELM_URL_V2 }} -o /usr/bin/rancher-helm-v2
+
+      - name: Download Rancher Tiller
+        run: sudo curl -sLf ${{ env.TILLER_URL }} -o /usr/bin/rancher-tiller
+
+      - name: Set permissions for Helm and Tiller
+        run: sudo chmod +x /usr/bin/rancher-helm-v2 /usr/bin/rancher-tiller
+
+      - name: Create symlink for Helm
+        run: sudo ln -s /usr/bin/rancher-helm-v2 /usr/bin/helm
+
+      - name: Create symlink for Tiller
+        run: sudo ln -s /usr/bin/rancher-tiller /usr/bin/tiller
+
+      - name: Initialize Helm
+        run: helm init -c --stable-repo-url https://charts.helm.sh/stable/
+
+      - name: Set Helm 3 URL
+        run: echo "HELM_URL_V3=https://get.helm.sh/helm-${{ env.HELM_VERSION }}-linux-${{ env.DAPPER_HOST_ARCH }}.tar.gz" >> $GITHUB_ENV
+
+      - name: Create temporary directory for Helm 3
+        run: sudo mkdir /usr/tmp
+
+      - name: Download and extract Helm 3
+        run: curl ${{ env.HELM_URL_V3 }} | sudo tar xvzf - --strip-components=1 -C /usr/tmp/
+
+      - name: Move Helm 3 binary
+        run: sudo mv /usr/tmp/helm /usr/bin/helm_v3
+
+      ## Helm Setup end
+
+      - name: Install kustomize
+        run: |
+          KUSTOMIZE_URL=https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/${{ env.KUSTOMIZE_VERSION }}/kustomize_${{ env.KUSTOMIZE_VERSION }}_linux_${{ env.DAPPER_HOST_ARCH }}.tar.gz
+          curl -sLf ${KUSTOMIZE_URL} | sudo tar -xzf - -C /usr/bin
+          sudo chmod +x /usr/bin/kustomize
+
+      ## Start K3s binaries
+
+      - name: Create temporary directory for binaries
+        run: sudo mkdir -p /tmp/k3s-binaries
+
+      - name: Copy all binaries from k3s image to temporary directory
+        run: |
+          docker run --rm --entrypoint "" -v /tmp/k3s-binaries:/host-bin rancher/k3s:${{ env.CATTLE_K3S_VERSION_DOCKER }} sh -c "
+            cp /bin/blkid /host-bin/ &&
+            cp /bin/bandwidth /host-bin/ &&
+            cp /bin/cni /host-bin/ &&
+            cp /bin/conntrack /host-bin/ &&
+            cp /bin/containerd /host-bin/ &&
+            cp /bin/containerd-shim-runc-v2 /host-bin/ &&
+            cp /bin/ethtool /host-bin/ &&
+            cp /bin/firewall /host-bin/ &&
+            cp /bin/ip /host-bin/ &&
+            cp /bin/ipset /host-bin/ &&
+            cp /bin/k3s /host-bin/ &&
+            cp /bin/losetup /host-bin/ &&
+            cp /bin/pigz /host-bin/ &&
+            cp /bin/runc /host-bin/ &&
+            cp /bin/which /host-bin/ &&
+            cp /bin/aux/xtables-legacy-multi /host-bin/"
+
+      #- name: List copied binaries
+        #run: ls -l /tmp/k3s-binaries
+
+      - name: Copy blkid binary to its place
```

Comment:
Would it make sense to have only one copy step in order to reduce noise?

Reply:
Squashed them into a single step.